frontend/src/InputForm.js

```diff
@@ -155,76 +164,150 @@ class InputForm extends React.Component {
+            : lectureSection)})
+        : course)
+
+    this.setState(state => ({
+      ...this.state,
+      courses: state.courses.map((course, courseIndex) =>
+        courseIndex === courseID
+          ? ({ ...course, lectureSections: state.courses[courseIndex].lectureSections.map((lectureSection, sectionIndex) =>
+            sectionIndex === sectionID
+              ? (
+                days.length > 0 
+                  ? ({ ...lectureSection, days: days})
+                  : ({ ...lectureSection, days: state.courses[courseIndex].lectureSections[sectionIndex].days.concat(day)})
+              )
+              : lectureSection)})
+          : course)
+    }))
+  }
+
+  updateLabSectionProfessor(courseID, sectionID, sectionProfessor) {
     this.setState(state => ({
       ...this.state,
       courses: state.courses.map((course, courseIndex) =>
         courseIndex === courseID
-          ? ({ ...course, sections: state.courses[courseIndex].sections.map((section, sectionIndex) =>
+          ? ({ ...course, labSections: state.courses[courseIndex].labSections.map((labSection, sectionIndex) =>
             sectionIndex === sectionID
-              ? ({...section, professor: sectionProfessor})
-              : section)})
+              ? ({...labSection, professor: sectionProfessor})
+              : labSection)})
           : course)
     }))
   }
 
-  updateSectionStartTime(courseID, sectionID, sectionStartTime) {
+  updateLabSectionStartTime(courseID, sectionID, sectionStartTime) {
     this.setState(state => ({
       ...this.state,
       courses: state.courses.map((course, courseIndex) =>
         courseIndex === courseID
-          ? ({ ...course, sections: state.courses[courseIndex].sections.map((section, sectionIndex) =>
+          ? ({ ...course, labSections: state.courses[courseIndex].labSections.map((labSection, sectionIndex) =>
             sectionIndex === sectionID
-              ? ({...section, startTime: sectionStartTime})
-              : section)})
+              ? ({...labSection, startTime: sectionStartTime})
+              : labSection)})
           : course)
     }))
   }
 
-  updateSectionEndTime(courseID, sectionID, sectionEndTime) {
+  updateLabSectionEndTime(courseID, sectionID, sectionEndTime) {
     this.setState(state => ({
       ...this.state,
       courses: state.courses.map((course, courseIndex) =>
         courseIndex === courseID
-          ? ({ ...course, sections: state.courses[courseIndex].sections.map((section, sectionIndex) =>
+          ? ({ ...course, labSections: state.courses[courseIndex].labSections.map((labSection, sectionIndex) =>
             sectionIndex === sectionID
-              ? ({...section, endTime: sectionEndTime})
-              : section)})
+              ? ({...labSection, endTime: sectionEndTime})
+              : labSection)})
           : course)
     }))
   }
 
-  updateSectionDays(courseID, sectionID, day) {
+  updateLabSectionDays(courseID, sectionID, day) {
```

Comment:
I just noticed the nested ternary operators in this file. I'd really recommend changing these to if-else statements - they are very difficult to follow as nested ternary operators.

Reply:
Yeah I see that it did get more unreadable when nested, I'll clear that up!